asyncapi/v2/validation.go

```diff
@@ -0,0 +1,60 @@
+package v2
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/asyncapi/event-gateway/asyncapi"
+	"github.com/asyncapi/event-gateway/proxy"
+	"github.com/xeipuuv/gojsonschema"
+)
+
+func FromDocJSONSchemaMessageValidator(doc asyncapi.Document) (proxy.MessageValidator, error) {
+	channels := doc.ApplicationSubscribableChannels()
+	messageSchemas := make(map[string]gojsonschema.JSONLoader)
+	for _, c := range channels {
+		for _, o := range c.Operations() {
+			if !o.IsApplicationSubscribing() {
+				continue
+			}
+
+			if len(o.Messages()) == 0 {
+				return nil, fmt.Errorf("can not generate message validation for operation %s. Reason:. Operation has no message. This is totally unexpected", o.ID())
+			}
+
+			var payload asyncapi.Schema
+			var messageNames string
+			if len(o.Messages()) > 1 {
+				// Meaning message payload is a Schema containing several payloads as `oneOf`.
+				// Generating back just one Schema adding all payloads to oneOf field.
```

Comment:
But be aware that if we will have support for `messageID` then it should be changed to single payload :) Maybe add this info in the comment?

Reply:
won't `messageID` be, if so, in version 3.x.x of the spec? I know it does not break anything anyway.

Reply:
Probably in the `3.X.X` or maybe in the `2.2.X`. It doesn't break, but as we had discussion, you merge here the payloads from all messages (from `oneOf`), so at the moment you haven't information which message do you handle - https://github.com/asyncapi/event-gateway/issues/10 :)

Reply:
Once we add messageID, we will need to add support obviously. So I think it's a thing of the future, a future that it is not so close as https://github.com/asyncapi/spec/issues/458 is in RFC1 status. 
https://github.com/asyncapi/parser-go is not even supporting `2.1.0` yet.